client/src/redux/sagas/buildSetSaga.js

```diff
@@ -1,16 +1,67 @@
-import { takeEvery, call, put } from 'redux-saga/effects';
-import { BUILD_DATA } from '../constants';
-import { fetchBuildSets } from '../api';
-import { setBuildSets, setBuildSetsError } from '../actions';
+import { takeEvery, call, put, select } from 'redux-saga/effects';
+import { COMMITS } from '../constants';
+import { fetchBuildSets, fetchBuildReq, fetchBuilds } from '../api';
+import { setBuildSetsError, setBuilds } from '../actions';
+
+const commitSha = state => state.commits;
+function buildStr(builds) {
+ return builds
+  .map(build => 'buildrequestid__contains=' + build.buildrequestid)
+  .join('&');
+}
+
+function getBuildSetReqString(commits, buildData) {
+ return commits
+  .map(commit =>
+   buildData.filter(bd => bd.sourcestamps[0].revision === commit.sha)
+  )
+  .flat()
+  .map(bd => 'buildsetid__contains=' + bd.bsid)
+  .join('&');
+}
+
 function* handleBuildsLoad() {
  try {
   const buildSets = yield call(fetchBuildSets);
-  yield put(setBuildSets(buildSets));
+  const commits = yield select(commitSha);
```

Comment:
A few comments about what's happening in this function would be nice. Especially about sequence of getting build data from commit hash